picard/ui/coverartbox.py

```diff
@@ -545,6 +550,71 @@ def load_remote_image(self, url, data):
             log.warning("Can't load image: %s", e)
             return
 
+    def delete_cover_art(self):
+        if not self.item or not self.item.metadata.images:
+            if not self.item.orig_metadata.images:
+                return
+
+        cover_art_list = [image.source or _("Unnamed Cover Art") for image in self.item.metadata.images]
+
+        selected_item, ok_pressed = QtWidgets.QInputDialog.getItem(self, _("Delete Cover Art"),
+                                                                    _("Select the cover art image to delete:"),
+                                                                    cover_art_list, 0, False)
+        if ok_pressed:
+            selected_image_index = cover_art_list.index(selected_item)
+            selected_image = self.item.metadata.images[selected_image_index]
+            try:
+                self.delete_cover_art_for_item(self.item, selected_image)
+            except CoverArtImageError as e:
+                log.error("Can't delete image: %s", e)
+                return
+
+    def delete_cover_art_for_item(self, item, selected_image):
+        metadata = item.metadata
+        if not metadata or not metadata.images:
+            return
+
+        debug_info = "Deleting %r from %r"
+
+        if isinstance(item, Album):
+            item.enable_update_metadata_images(False)
+            for track in item.tracks:
+                track.enable_update_metadata_images(False)
+                image_delete(track, selected_image)
+            for file in item.iterfiles():
+                image_delete(file, selected_image)
+                file.update(signal=False)
+            for track in item.tracks:
+                track.enable_update_metadata_images(True)
+            item.enable_update_metadata_images(True)
+            item.update(update_tracks=False)
+        elif isinstance(item, FileListItem):
+            parents = set()
+            item.enable_update_metadata_images(False)
+            image_delete(item, selected_image)
+            for file in item.iterfiles():
+                for parent in iter_file_parents(file):
+                    parent.enable_update_metadata_images(False)
+                    parents.add(parent)
+                image_delete(file, selected_image)
+                file.update(signal=False)
+            for parent in parents:
+                image_delete(parent, selected_image)
+                parent.enable_update_metadata_images(True)
+                if isinstance(parent, Album):
+                    parent.update(update_tracks=False)
+                else:
+                    parent.update()
+            item.enable_update_metadata_images(True)
+            item.update()
+        elif isinstance(item, File):
+            image_delete(item, selected_image)
+            item.update()
+        else:
+            debug_info = "Dropping %r to %r is not handled"
```

Comment:
For consistency in terminology, would it be more appropriate to change this to something like "Unable to delete %r from %r"?